### DISCUSSION TITLE
REST Api fossology 4.4

### DISCUSSION THREAD

**Discussion by YanickNoblanc** [fossology/fossology#2726] [Q&A] [answered]
The REST API documentation available here :  https://github.com/fossology/fossology/wiki/FOSSology-REST-API Seems to be deprecated for fossology 4.4.
Specificaly 'c. Upload from URL' syntax described is not working with fossology 4.4.0.35.
Where can we get an updated REST documentation ?

**Comment by YanickNoblanc** [comment]
Get message : 
Require location object if uploadType != file

**Comment by YanickNoblanc** [comment]
Full curl command :
curl --connect-timeout "${CURL_TIMEOUT}" -k -s -S -L -X POST "${FOSSURL}/repo/api/v1/uploads" \
       -H "folderId: ${FOLDERID}" \
       -H 'uploadDescription: created by REST' \
       -H 'public: public' \
       -H 'Content-Type: application/json' \
       -H 'uploadType: url' \
       -H "Authorization: Bearer ${FOSSTOKEN}" \
       --data-raw "{
        \"url\": \"${PKGURL}\",
        \"name\": \"${PKGFILENAME}\"
        }"

**Comment by dvjsharma** [comment, marked as answer]
Hey @YanickNoblanc,

You can access the v1 API documentation in [openapi.yaml](https://github.com/fossology/fossology/blob/master/src/www/ui/api/documentation/openapi.yaml). This file contains the most updated documentation, including any recent changes to the API code. It can also be imported into [Swagger Editor](http://editor.swagger.io/) for a more user-friendly interface. The editor offers CURL commands for your reference. Just send a request to view the command.

![image](https://github.com/fossology/fossology/assets/119073504/ed0f7aeb-6bb8-4f98-9b3e-0f1fd2ab88b0)
_a `GET` request to `/uploads` endpoint_

**Comment by YanickNoblanc** [comment]
Thanks a lot

**Comment by YanickNoblanc** [comment]
Doc available
